src/nmodl/visitors/matexp_visitor.cpp

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2025 David McDougall
+ * See the top-level LICENSE file for details.
+ *
+ * SPDX-License-Identifier: Apache-2.0
```

Comment:
Can you clarify the license situation?

[NEURON is BSD-licensed](https://github.com/neuronsimulator/nrn?tab=License-1-ov-file) but the internet [seems to say](https://github.com/neuronsimulator/nrn?tab=License-1-ov-file) that Apache 2.0 is a more restrictive license.

If this is entirely your code, any chance you'd be willing to contribute it as BSD-3 licensed instead?

Reply:
I would happy to change the license.
I copied the license from the other files in the NMODL project directory, which are all Apache-2 licensed.

Reply:
Okay, I changed the license on my new files to "BSD-3-Clause".

However, the rest of nmodl still uses the Apache-2.0 license which it inherited from the Blue Brain Project: https://github.com/BlueBrain/nmodl?tab=Apache-2.0-1-ov-file